Core/include/Acts/Propagator/Navigator.hpp

```diff
@@ -696,129 +661,259 @@ class Navigator {
   }
 
  private:
-  /// @brief Resolve compatible surfaces
+  ///@brief NextTarget helper function for gen3 navigation
```

Comment:
this function now handles both, Gen1 and Gen3 now, no?

should we split this into `tryGetNextTargetGen1` and `tryGetNextTargetGen3` but keep `tryGetNextTarget` to branch into one of them?

Reply:
as you prefer - at least the documentation should be changed IMO